### DISCUSSION TITLE
Using @Value inside Changelog

### DISCUSSION THREAD

**Discussion by hashamrasheed** [flamingock/mongock#41] [Q&A] [answered]
Hi!,

I'm trying to use external properties inside my ChangeLog. The accessKey field is always null:

```
@ChangeLog(order = "10")
public class IconsUploader {
  @Value("${client.accessKey}")
  private String accessKey;
}
```

I'm using @EnableMongock approach with Sprint Boot.

**Comment by dieppa** [comment]
Hello @hrasheed90 ,

thank you for raising this discussion. Although this is in our roadmap, it's still not supported by Mongock.

Meanwhile, as a workaround, I suggest to use Environment and something like this: `String accessKey = env.getProperty("client.accessKey");`. This should work

**Reply by hashamrasheed** [reply]
Thanks for the reply @dieppa . I tried autowiring environment too, but it's always null:

```
@ChangeLog(order = "10")
public class IconsUploader {
  @Autowired
 private Environment env;

 @ChangeSet(order = "1", id = "testUpload", author = "hasham", runAlways = true)
    public void testUploadS3() {
     if(env == null) {
    LOG.error("Environment is null");
}
}
```

**Comment by dieppa** [comment, marked as answer]
Hello, that makes sense. Injections in Mongock are performed via method parameter. Since version 5, it also supports constructor parameters 😉 

Which I also realise now, you were also injecting `@Value` as field injection. That wouldn't have worked neither

**Reply by hashamrasheed** [reply]
Could you please give me an example of injection via method param? I believe I received an error saying something like "Incorrect or unexpected parameter"

**Reply by dieppa** [reply]
```java
 @ChangeSet(order = "1", id = "testUpload", author = "hasham", runAlways = true)
    public void testUploadS3(Environment) {
     if(env == null) {
    LOG.error("Environment is null");
}
```

Mongock injects a proxy to make sure injections are synchronised. In this case, `env` doesn't need to be synchronised, so you can add `@NonLockGuarded` annotation like this and you will obtain the raw environment :
```java
 @ChangeSet(order = "1", id = "testUpload", author = "hasham", runAlways = true)
    public void testUploadS3(@NonLockGuarded  Environment) {
     if(env == null) {
    LOG.error("Environment is null");
}
```

**Reply by hashamrasheed** [reply]
Thank you! Let me try that and get back to you.

**Reply by hashamrasheed** [reply]
It worked like a charm - thank you!
